clang-tools-extra/clang-doc/HTMLMustacheGenerator.cpp

```diff
@@ -0,0 +1,523 @@
+  for (const MemberTypeInfo &Member : I.Members ) {
+    Value MemberValue = Object();
+    MemberValue.getAsObject()->insert({"Name", Member.Name});
+    MemberValue.getAsObject()->insert({"Type", Member.Type.Name});
+    if (!Member.Description.empty()) {
+      Value ArrDesc = Array();
+      for (const CommentInfo& Child : Member.Description) 
+        ArrDesc.getAsArray()->emplace_back(extractValue(Child));
+      MemberValue.getAsObject()->insert({"MemberComments", ArrDesc });
+    }
+    
+    if (Member.Access == AccessSpecifier::AS_public)
+      PublicMembers.getAsArray()->emplace_back(MemberValue);
+    else if (Member.Access == AccessSpecifier::AS_protected)
+      ProtectedMembers.getAsArray()->emplace_back(MemberValue);
+    else if (Member.Access == AccessSpecifier::AS_private)
+      PrivateMembers.getAsArray()->emplace_back(MemberValue);
+  }
+  if (!PublicMembers.getAsArray()->empty())
+    RecordValue.insert({"PublicMembers", Object{{"Obj", PublicMembers}}});
+  if (!ProtectedMembers.getAsArray()->empty())
+    RecordValue.insert({"ProtectedMembers", Object{{"Obj", ProtectedMembers}}});
+  if (!PrivateMembers.getAsArray()->empty())
+    RecordValue.insert({"PrivateMembers", Object{{"Obj", PrivateMembers}}});
+  
+  return RecordValue;
+}
+
+void setupTemplateValue(const ClangDocContext &CDCtx, Value &V, Info *I) {
+  V.getAsObject()->insert({"ProjectName", CDCtx.ProjectName});
+  Value StylesheetArr = Array();
+  auto InfoPath = I->getRelativeFilePath("");
+  SmallString<128> RelativePath = computeRelativePath("", InfoPath);
+  for (const auto &FilePath  : CDCtx.UserStylesheets) {
+    SmallString<128> StylesheetPath = RelativePath;
+    llvm::sys::path::append(StylesheetPath,
+                            llvm::sys::path::filename(FilePath));
+    llvm::sys::path::native(StylesheetPath, llvm::sys::path::Style::posix);
+    StylesheetArr.getAsArray()->emplace_back(StylesheetPath);
+  }
+  V.getAsObject()->insert({"Stylesheets", StylesheetArr});
+  
+  Value ScriptArr = Array();
+  for (auto Script : CDCtx.JsScripts) {
+    SmallString<128> JsPath = RelativePath;
+    llvm::sys::path::append(JsPath, llvm::sys::path::filename(Script));
+    ScriptArr.getAsArray()->emplace_back(JsPath);
+  }
+  V.getAsObject()->insert({"Scripts", ScriptArr});
+}
+ 
+
+llvm::Error
+MustacheHTMLGenerator::generateDocForInfo(Info *I, llvm::raw_ostream &OS,
+                                          const ClangDocContext &CDCtx) {
+  switch (I->IT) {
+  case InfoType::IT_namespace: {
+    Value V = extractValue(*static_cast<clang::doc::NamespaceInfo *>(I), CDCtx);
+    setupTemplateValue(CDCtx, V, I);
+    OS << NamespaceTemplate->render(V);
+    break;
+  }
+  case InfoType::IT_record: {
+    Value V = extractValue(*static_cast<clang::doc::RecordInfo *>(I), CDCtx);
+    setupTemplateValue(CDCtx, V, I);
+    // Serialize the JSON value to the output stream in a readable format.
+    llvm::outs() << "Visit: " << I->Name << "\n";
```

Comment:
Won't this be pretty verbose? Should this be wrapped in LLVM_DEBUG or something instead?